File: Java/fibonacci.java
public class Main {
  static long fib(int n) {
    if (n <= 1) return n;
    return fib(n - 1) + fib(n - 2);
  }
  
  public static void main(String[] args) {
    fib(10); // returns 55
  }
}

File: Rust/fibonacci.rs
fn main() {
  fn fib(n: usize) -> usize {
    match n {
      0 => 0,
      1 => 1,
      _ => fib(n - 1) + fib(n - 2)
    }
  }

  fib(10); // returns 55
}
